File: R/zcjs.R
#' @import htmlwidgets
#' @export
zcjs <- function(url,
                 plotMethod = "ploty",
                 width = NULL,
                 height = NULL,
                 x_range = "ms",
                 x_compress = FALSE,
                 y_range = "nonzero",
                 y_fixed = TRUE) {

  # create a list that contains the settings
  settings <- list(
    x_range = x_range,
    x_compress = x_compress,
    y_range = y_range,
    y_fixed = y_fixed
  )

  # pass the data and settings using 'x'
  x <- list(
    url = url,
    settings = settings
  )

  # create the widget
  htmlwidgets::createWidget("zcjs", x, width = width, height = height)
}

File: inst/htmlwidgets/zcjs.js
HTMLWidgets.widget({

  name: "zcjs",

  type: "output",

  factory: function(el, width, height) {
    // create our sigma object and bind it to the element
    var zc = new ZCJS(el.id);

    return {
      renderValue: function(x) {

        if (x.settings.x_compress) {
          zc.x_compress = true;
        }
        zc.setURL(x.url);
      },

      resize: function(width, height) {
        // forward resize on to sigma renderers
        /*
        for (var name in sig.renderers)
          sig.renderers[name].resize(width, height);
          */
      },

      // Make the zcjs object available as a property on the widget
      // instance we're returning from factory(). This is generally a
      // good idea for extensibility--it helps users of this widget
      // interact directly with zcjs, if needed.
      s: zc
    };
  }
});
